compiler-plugin/src/main/kotlin/arrow/meta/plugins/proofs/phases/ir/ProofsIrCodegen.kt

```diff
@@ -173,45 +171,40 @@ class ProofsIrCodegen(
       expression
     }
 
-  private fun CompilerContext.insertExtensionSyntaxCall(expression: IrCall) {
+  private fun CompilerContext.insertExtensionSyntaxCall(expression: IrCall) = irUtils.run {
     val valueType = expression.dispatchReceiver?.type?.toKotlinType()
       ?: expression.extensionReceiver?.type?.toKotlinType()
       ?: (if (expression.valueArgumentsCount > 0) expression.getValueArgument(0)?.type?.toKotlinType() else null)
     val targetType =
-      (expression.symbol.owner.descriptor.dispatchReceiverParameter?.containingDeclaration as? FunctionDescriptor)?.dispatchReceiverParameter?.type
-        ?: expression.symbol.owner.descriptor.extensionReceiverParameter?.type
-        ?: expression.symbol.owner.descriptor.valueParameters.firstOrNull()?.type
+      (expression.unsubstitutedDescriptor.dispatchReceiverParameter?.containingDeclaration as? FunctionDescriptor)?.dispatchReceiverParameter?.type
+        ?: expression.unsubstitutedDescriptor.extensionReceiverParameter?.type
+        ?: expression.substitutedValueParameters.firstOrNull()?.second
```

Comment:
This is what we were looking for? is it returning the resolved type??

Reply:
that is only a shorter version for the other expression